docs/security/self-assessment.md

```diff
@@ -0,0 +1,167 @@
+# Self-assessment
+This document details the design goals and security implications of KubeEdge to aid in the security assessment by CNCF TAG-Security.
+
+# Self-assessment outline
+
+## Table of contents
+
+* [Metadata](#metadata)
+  * [Security links](#security-links)
+* [Overview](#overview)
+  * [Background](#background)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|                   |                                                              |
+| ----------------- | ------------------------------------------------------------ |
+| Software          | https://github.com/kubeedge/kubeedge                         |
+| Website           | https://kubeedge.io                                          |
+| Security Provider | No                                                           |
+| Languages         | Go                                                           |
+| SBOM              | Check [go.mod](https://github.com/kubeedge/kubeedge/blob/master/go.mod) for libraries, packages, versions used by the project |
+
+### Security links
+
+| Doc                          | url                                                          |
+| ---------------------------- | ------------------------------------------------------------ |
+| Security file                | [Security policy](https://github.com/kubeedge/kubeedge/security/policy) |
+| Default and optional configs | [cloudcore](https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/componentconfig/cloudcore/v1alpha1/default.go)<br />[edgecore](https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/componentconfig/edgecore/v1alpha1/default.go) |
+
+## Overview
```

Comment:
Some basic information seems out of date, please check whether the orginal reference is up to or out of date.